### PR TITLE
Add rank eval documentation

### LINKED ISSUE #1268: [DOC] API - add documentation for rank.eval
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [X] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 
API - add documentation for rank.eval

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

### PR DESCRIPTION
**Pull request by ariamarble**
### Description

Adds rank eval documentation

### Issues Resolved

#1268 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by ariamarble** [comment]
tech review and testing completed

File: _opensearch/rest-api/rank-eval.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+The [rank]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/rank/) eval endpoint allows you to evaluate the quality of ranked search results.
+
+## Path and HTTP methods
+
+```
+GET <index_name>/_rank_eval 
+POST <index_name>/_rank_eval
+```
+
+## URL parameters
```
Comment (kolchfa-aws) [inline _opensearch/rest-api/rank-eval.md:19]: Could we make this heading "Query parameters"?

File: _opensearch/rest-api/rank-eval.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+URL Parameters are optional.
```
Comment (kolchfa-aws) [inline _opensearch/rest-api/rank-eval.md:21]: Query parameters

File: _opensearch/rest-api/rank-eval.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+Parameter | Data Type | Description
+:--- | :---  | :---
+[`ignore_unavailable`](#ignore_unavailable) | boolean | Defaults to `false`. When set to `false` the response body will return an error if an index is closed or missing.
+[`allow_no_indices`](#allow_no_indices) | boolean | Defaults to `true`. When set to `false` the response body will return an error if a wildcard expression points to an indices that are closed or missing.
+[`expand_wildcards`](#expand_wildcards) | enum | Expand wildcard expressions for indicies that are open, closed, or both.
```
Comment (kolchfa-aws) [inline _opensearch/rest-api/rank-eval.md:27]: I get that it is an enum, but for users' purposes I think it makes more sense to make this a string and enumerate the options with explanations.

File: _opensearch/rest-api/rank-eval.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+POST <index_name>/_rank_eval
+```
+
+## URL parameters
+
+URL Parameters are optional.
+
+Parameter | Data Type | Description
+:--- | :---  | :---
+[`ignore_unavailable`](#ignore_unavailable) | boolean | Defaults to `false`. When set to `false` the response body will return an error if an index is closed or missing.
```
Comment (kolchfa-aws) [inline _opensearch/rest-api/rank-eval.md:25]: Are these links going to be in the same file?

File: _opensearch/rest-api/rank-eval.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+[`allow_no_indices`](#allow_no_indices) | boolean | Defaults to `true`. When set to `false` the response body will return an error if a wildcard expression points to an indices that are closed or missing.
+[`expand_wildcards`](#expand_wildcards) | enum | Expand wildcard expressions for indicies that are open, closed, or both.
+[`search_type`](#search_type) | enum | Set search type.
+
+## Request body parameters
```
Comment (kolchfa-aws) [inline _opensearch/rest-api/rank-eval.md:30]: Request fields

File: _opensearch/rest-api/rank-eval.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+Request body must contain field type parameters.
```
Comment (kolchfa-aws) [inline _opensearch/rest-api/rank-eval.md:32]: So, in other words, are all of the fields required?

File: _opensearch/rest-api/rank-eval.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+[`ratings`](#ratings) | Document relevance score.
+[`k`](#k) | The number of documents returned per query. Default is set to 10.
+[`relevant_rating_threshold`](#relevant_rating_threshold) | The threshold that documents are considered relevant. Default is set to 1.
+[`normalize`](#normalize) | Discounted cumulative gain will be calculated when set to `true`.
+[`maximum_relevance`](#maximum_relevance) | Sets the maximum relevance score when using the expected reciprocal rank metric.
+[`ignore_unlabeled`](#ignore_unlabeled) | Defaults to `false`. Unlabeled documents are ignored when set to `true`. 
+[`template_id`](#template_id) | Template ID.
+[`params`](#params) | Parameters used in the template.
+
+## Examples
```
Comment (kolchfa-aws) [inline _opensearch/rest-api/rank-eval.md:47]: We can just say Sample request (h4) and Sample response (h4) to match our API style guide.

File: _opensearch/rest-api/rank-eval.md (hunk 4 above)
Comment (ariamarble) [reply _opensearch/rest-api/rank-eval.md:25]: yeah, I was on the fence with this as I was formatting so I think I'll just remove these are most other pages don't match up

File: _opensearch/rest-api/rank-eval.md (hunk 6 above)
Comment (ariamarble) [reply _opensearch/rest-api/rank-eval.md:32]: I'll clarify

File: _opensearch/rest-api/rank-eval.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+## Path and HTTP methods
+
+```
+GET <index_name>/_rank_eval 
+POST <index_name>/_rank_eval
+```
+
+## Query parameters
+
+Query Parameters are optional.
```
Comment (kolchfa-aws) [inline _opensearch/rest-api/rank-eval.md:21]: parameters with lowercase 'p'

File: _opensearch/rest-api/rank-eval.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+Parameter | Data Type | Description
+:--- | :---  | :---
+ignore_unavailable | boolean | Defaults to `false`. When set to `false` the response body will return an error if an index is closed or missing.
+allow_no_indices | boolean | Defaults to `true`. When set to `false` the response body will return an error if a wildcard expression points to an indices that are closed or missing.
+expand_wildcards | string | Expand wildcard expressions for indices that are `open`, `closed`, `hidden`, `none`, or `all`.
+search_type | enum | Set search type.
+
+## Request fields
+
+The request body must contain at least one field type parameter.
```
Comment (kolchfa-aws) [inline _opensearch/rest-api/rank-eval.md:32]: at least one parameter

**Review by kolchfa-aws — APPROVED** [review]
LGTM

**Review by alicejw-aws — APPROVED** [review]
Looks great! good work Aria

File: _opensearch/rest-api/rank-eval.md [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+POST <index_name>/_rank_eval
+```
+
+## Query parameters
+
+Query parameters are optional.
+
+Parameter | Data Type | Description
+:--- | :---  | :---
+ignore_unavailable | boolean | Defaults to `false`. When set to `false` the response body will return an error if an index is closed or missing.
```
Comment (natebower) [inline _opensearch/rest-api/rank-eval.md:25]: Capitalize all the items in the "Data Type" column.

File: _opensearch/rest-api/rank-eval.md [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+allow_no_indices | boolean | Defaults to `true`. When set to `false` the response body will return an error if a wildcard expression points to an indices that are closed or missing.
```
Comment (natebower) [inline _opensearch/rest-api/rank-eval.md:26]: In the last sentence, it should either be "...points to an index that is closed or missing" or "...points to indexes that are closed or missing."

File: _opensearch/rest-api/rank-eval.md [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+expand_wildcards | string | Expand wildcard expressions for indices that are `open`, `closed`, `hidden`, `none`, or `all`.
```
Comment (natebower) [inline _opensearch/rest-api/rank-eval.md:27]: Change "indices" to "indexes".

File: _opensearch/rest-api/rank-eval.md [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+search_type | enum | Set search type.
```
Comment (natebower) [inline _opensearch/rest-api/rank-eval.md:28]: Per Fanit: We can't just say 'enum': we need to say "string" and then list all options in the description.

File: _opensearch/rest-api/rank-eval.md [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+
+The request body must contain at least one parameter.
+
+Field Type | Description
+:--- | :---  
+id | Document or template ID.
+requests | Set multiple search requests within the request field section.
+ratings | Document relevance score.
+k | The number of documents returned per query. Default is set to 10.
+relevant_rating_threshold | The threshold that documents are considered relevant. Default is set to 1.
```
Comment (natebower) [inline _opensearch/rest-api/rank-eval.md:40]: The description needs a little revision. Do we mean "The threshold at which documents are considered relevant"?

**Review by natebower — COMMENTED** [review]
@ariamarble Please see my comments and let me know if you have any questions. Thanks!

**Pull request merged**